app/views/favorites/of_user.ctp

```diff
@@ -43,45 +55,56 @@ $numberOfSentences = count($favorites);
-    if ($numberOfSentences > 0) {
-        $type = 'mainSentence';
-        $parentId = null;
-        $withAudio = false;
-        $ownerName = null;
-        foreach ($favorites as $favorite) {
-            if (empty($favorite['Sentence']['text'])) {
-                $sentenceId = $favorite['Favorite']['favorite_id'];
-                $linkToSentence = $html->link(
-                    '#'.$sentenceId,
-                    array(
-                        'controller' => 'sentences',
-                        'action' => 'show',
-                        $sentenceId
-                    )
-                );
+    if ($userExists === false) {
+        $commonModules->displayNoSuchUser($username, $backLink);
+    } else {
+        echo '<h2>';
+        echo $title . ' ('. $numberOfSentences.')';
+        echo '</h2>';
+        if ($numberOfSentences > 0) {
+
+            $paginationUrl = array($username);
+            $pagination->display($paginationUrl);
+
+            $type = 'mainSentence';
+            $parentId = null;
+            $withAudio = false;
+            $ownerName = null;
+            foreach ($favorites as $favorite) {
+                if (empty($favorite['Sentence']['text'])) {
+                    $sentenceId = $favorite['Favorite']['favorite_id'];
+                    $linkToSentence = $html->link(
+                        '#'.$sentenceId,
+                        array(
+                            'controller' => 'sentences',
+                            'action' => 'show',
+                            $sentenceId
+                        )
+                    );
 
-                echo $html->div('sentence deleted',
-                    format(
-                        __('Sentence {id} has been deleted.', true),
-                        array('id' => $linkToSentence)
-                    )
-                );
-            } else {
-                $sentences->displayGenericSentence(
-                    $favorite['Sentence'],
-                    $favorite['Sentence']['Transcription'],
-                    $type,
-                    $parentId,
-                    $withAudio
-                );
+                    echo $html->div('sentence deleted',
+                        format(
+                            __('Sentence {id} has been deleted.', true),
+                            array('id' => $linkToSentence)
+                        )
+                    );
+                } else {
+                    $sentences->displayGenericSentence(
+                        $favorite['Sentence'],
+                        $favorite['Sentence']['Transcription'],
+                        $type,
+                        $parentId,
+                        $withAudio
+                    );
+                }
             }
+            $pagination->display($paginationUrl);
+
+        } else {
+            __('This user does not have any favorites.');
+            echo $html->link(__('Go back to previous page', true), $backLink);
```

Comment:
Yes, backlinks are out of the scope of this PR. You did right by imitating existing files, it’s just that that code happens to be old.

If you want to remove all the backlinks from the codebase, you can do it, but it’s not very fun and there are maybe more important things to do (although code maintenance is important too). It’s up to you.

@trang any comment about backlinks?